packages/kbn-optimizer/src/worker/bundle_ref_module.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore not typed by @types/webpack
+import Module from 'webpack/lib/Module';
+
+export class BundleRefModule extends Module {
+  public built = false;
+  public buildMeta?: any;
+  public buildInfo?: any;
+  public exportsArgument = '__webpack_exports__';
+
+  constructor(public readonly exportId: string) {
+    super('kbn/bundleRef', null);
+  }
+
+  libIdent() {
+    return this.exportId;
+  }
+
+  chunkCondition(chunk: any) {
+    return chunk.hasEntryModule();
+  }
+
+  identifier() {
+    return '@kbn/bundleRef ' + JSON.stringify(this.exportId);
+  }
+
+  readableIdentifier() {
+    return this.identifier();
+  }
+
+  needRebuild() {
+    return false;
+  }
+
+  build(_: any, __: any, ___: any, ____: any, callback: () => void) {
+    this.built = true;
+    this.buildMeta = {};
+    this.buildInfo = {};
+    callback();
+  }
+
+  source() {
+    return `
+      __webpack_require__.r(__webpack_exports__);
+      var ns = __kbnBundles__.get('${this.exportId}');
+      Object.defineProperties(__webpack_exports__, Object.getOwnPropertyDescriptors(ns))
+    `;
+  }
+
+  size() {
+    return 42;
```

Comment:
Was this a Douglas Adams placeholder, or intentional?

Reply:
Haha, intentionally lifted from https://github.com/webpack/webpack/blob/master/lib/ExternalModule.js#L375

Reply:
Yikes, ok.

Reply:
We should add a `@notice` for the Webpack license then. It's currently included since we ship Webpack, but that will be removed pretty soon.

Reply:
I'm not opposed, but we don't ship this module either.

Reply:
[https://github.com/elastic/kibana/pull/68986/commits/6f16b8729027c47a90950d6d1251555bdfe53bae ](https://github.com/elastic/kibana/pull/68986/commits/6f16b8729027c47a90950d6d1251555bdfe53bae)